core/domain/state_domain.py

```diff
@@ -918,8 +942,933 @@ def get_all_outcomes(self) -> List[Outcome]:
+                    # `IsEqualToOrdering` rule should not have empty values.
+                    if len(rule_spec.inputs['x']) <= 0:
+                        raise utils.ValidationError(
+                            f'The rule \'{rule_spec_index}\'of '
+                            f'answer group \'{ans_group_index}\', '
+                            f'having rule type IsEqualToOrdering '
+                            f'should not have empty values.'
+                        )
+                    # `IsEqualToOrdering` rule should always come before
+                    # `HasElementXAtPositionY` where element `X` is present
+                    # at position `Y` in `IsEqualToOrdering` rule.
+                    for ele in ele_x_at_y_rules:
+                        ele_position = ele['position']
+                        ele_element = ele['element']
+                        rule_choice = rule_spec.inputs['x'][
+                            ele_position - 1]
+
+                        for choice in rule_choice:
+                            if choice == ele_element:
+                                raise utils.ValidationError(
+                                    f'Rule - {rule_spec_index} of '
+                                    f'answer group {ans_group_index} '
+                                    f'will never be match '
+                                    f'because it is made redundant by the '
+                                    f'HasElementXAtPositionY rule above.'
+                                )
+                    # `IsEqualToOrdering` should always come before
+                    # `IsEqualToOrderingWithOneItemAtIncorrectPosition` when
+                    # they are off by one value.
+                    item_to_layer_idx = {}
+                    for layer_idx, layer in enumerate(rule_spec.inputs['x']):
+                        for item in layer:
+                            item_to_layer_idx[item] = layer_idx
+
+                    for ele in equal_ordering_one_at_incorec_posn:
+                        is_off_by_one_value = False
+                        for layer_idx, layer in enumerate(ele):
+                            for item in layer:
+                                if layer_idx != item_to_layer_idx[item]:
+                                    is_off_by_one_value = True
+                        if is_off_by_one_value:
+                            raise utils.ValidationError(
+                                f'Rule - {rule_spec_index} of answer '
+                                f'group {ans_group_index} will never '
+                                f'be match because it is made '
+                                f'redundant by the IsEqualToOrdering'
+                                f'WithOneItemAtIncorrectPosition '
+                                f'rule above.'
+                            )
+
+        choices = self.customization_args['choices'].value
+        if len(choices) < 2:
```

Comment:
Can be added via schema!

Reply:
Hey @DubeySandeep, We already have a schema validation for this part but instead of `atleast 2 length` it is `atleast 1` and have a comment explaining that we need to do it because for the default value we have to put it as empty strings. You can find the comment [here](https://github.com/oppia/oppia/blob/develop/extensions/interactions/DragAndDropSortInput/DragAndDropSortInput.py#L49). What should I do? Do I remove the current validation or not.
Thanks

Reply:
We have moved this check to schema now, right? (or we are still facing any issues?)

Reply:
Done